engines/registry/app/controllers/registry/registry_controller.rb

```diff
@@ -19,12 +19,23 @@ def authorize
     # Returns a Distribution Registry HTTP API V2 - compatible repository catalog as defined in
     # https://distribution.github.io/distribution/spec/api/#listing-repositories
     def catalog
+      raise 'Could not find system with current credentials' if @client&.account.nil?
+
       access_scope = AccessScope.parse('registry:catalog:*')
       repos = access_scope.allowed_paths(System.find_by(login: @client&.account))
       logger.debug("Returning #{repos.size} repos for client #{@client}")
 
       response.set_header('Docker-Distribution-Api-Version', REGISTRY_API_VERSION)
       render json: { repositories: repos }, status: :ok
+    rescue StandardError => e
```

Comment:
This would also catch any exception in the interim code - like a failure in JSON rendering. Doesn't guarantee the catalog search failed due to lack of authentication.

Reply:
Resolved in 3b832353.